modules/static/src/main/java/org/opencastproject/fsresources/StaticResourceServlet.java

```diff
@@ -101,28 +150,59 @@ public void activate(ComponentContext cc) {
         }
       }
     }
+    logger.debug("Authentication check enabled: {}", authRequired);
 
     if (StringUtils.isEmpty(distributionDirectory)) {
       throw new ConfigurationException("Distribution directory not set");
     }
     logger.info("Serving static files from '{}'", distributionDirectory);
   }
 
+  @Override
+  protected void doHead(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    logger.debug("Looking for static resource '{}'", req.getRequestURI());
+    String path = req.getPathInfo();
+    if (path == null || path.contains("..")) {
+      resp.sendError(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    if (authRequired && !isAuthorized(path)) {
+      resp.sendError(HttpServletResponse.SC_FORBIDDEN);
+      logger.debug("Not authorized");
+      return;
+    }
+
+    final File file = new File(distributionDirectory, path);
+    if (!file.isFile() || !file.canRead()) {
```

Comment:
If it's not a file then return a 404, but if it *is* a file but we can't read it (I'm assuming IO error, or filesystem ACLs) then we ought to either return a 503 (IO Error) or a 503/403 (filesystem ACL), IMO.

Reply:
I realize that this is confusing but it's intentional to properly work with the [nginx auth_request](http://nginx.org/en/docs/http/ngx_http_auth_request_module.html) which requires a 2xx, 401 or 403 response code. I add adding an example and explanation for this to my todo list :-)